### ISSUE TITLE
"NoMethodError: undefined method `[]' for nil:NilClass" in media_info.rb:22

### ISSUE THREAD

**Issue by christiangenco** [Jesus/dropbox_api#35]
I'm having the hardest time tracking down a bug in using this library for `save_url` and `save_url_check_job_status`.

I can't recreate it, but while it's running in background jobs (with sidekiq), `save_url_check_job_status` will throw `NoMethodError: undefined method '[]' for nil:NilClass` tracing to `media_info.rb:22`:

https://github.com/Jesus/dropbox_api/blob/163f3a54aabb0b943728d303db09d22670a825f8/lib/dropbox_api/metadata/media_info.rb#L22

So sometimes when `save_url_check_job_status` is called, `data['metadata']` is nil. Is this a problem with this library, or could it be that I'm calling it in a strange way?

**Comment by Jesus** [comment]
Hi,

thanks for reporting this. The issue has been fixed and I'll release a new gem version today.

It was difficult to track down because you had to catch it in the short lapse of time Dropbox needs to generate the media metadata.

The problem was that we were getting a pending tag as metadata when it hadn't been generated yet. This is specified in the documentation so it's a pity we missed it when we implemented this.

Here's the test that would catch this problem: https://github.com/Jesus/dropbox_api/blob/master/spec/metadata/file_spec.rb#L42-L55
